### ISSUE TITLE
PIN: Timeouts or crashes looting containers

### ISSUE THREAD

**Issue by noobanidus** [noobanidus/Lootr#172]
# TLDR; Install [Async Locator](https://beta.curseforge.com/minecraft/mc-mods/async-locator) OR [Async Locator Refined](https://www.curseforge.com/minecraft/mc-mods/async-locator-refined) 

# Identifying the source

The source may be a **Treasure Map in a container** or **something else**. To determine which, check your crash report. The presence of "StructureCheck", "ExplorationMapFunction", "LootItemFunction" or "LootPool" will indicate that this is a **Treasure Map issue**, so please check the next section.

If your crash doesn't contain this, **please create a new issue rather than replying to this**.

# Treasure Map containers

Treasure maps function by searching for a specific structure (such as a buried treasure, common in Shipwreck treasure chests) within a radius. This *generally* causes lag regardless in Vanilla, and has the potential to cause your server to crash with a "Watchdog" crash, where a tick takes too long.

***As Lootr uses identical methods to generate loot to Vanilla, this isn't something that Lootr can solve.*** 

***This is a Vanilla bug.***

The issue can be exacerbated by mods that increase the search radius, although I'm unaware of any mods that currently do this generally.

## Recommended solution (Forge & Fabric)

The [Async Locator](https://beta.curseforge.com/minecraft/mc-mods/async-locator) mod can mitigate this by causing biome and structure look-up to take place off-thread, meaning that you will never experience the initial lag. Unfortunately, this mod is only available from 1.18 onwards. 

## Terralith-specific

**If you are running Terralith specifically**: This issue may happen when structures associated with biomes do not generate within the search ratio. [Please see this issue on the FTB tracker for more information](https://github.com/FTBTeam/FTB-Modpack-Issues/issues/809#issuecomment-1136167755). There's a specific option for Terralith to fix this:

`/scoreboard players set %DISABLE_MAP tr.disable_maps 1`

Please note, I'm not 100% sure if this is the most recently suggested solution.

## Reference issues

#100 
#152 
#136 
#170 
https://github.com/FTBTeam/FTB-Modpack-Issues/issues/809
https://github.com/EnigmaticaModpacks/Enigmatica8/issues/320

**Comment by pokesmells** [comment]
Ah so this is what the issue was, I legit thought I was tripping when I had this issue!

Only occurs like you said with a chest with treasure maps, and I don't necessarily get a crash but it takes AGES to stop freezing.

Mod Version: lootr-1.19-0.4.25.64.jar
OS: Windows 11
Fabric Version: 0.14.15 - fabric-api-0.73.2+1.19.2

This has occurred on a single-player setting, MANY times, I also do not crash so no logs indicating what's happening, I also do not have Terralith. 

This may be available in the json, but I'm assuming there's a way to disable treasure map spawns on the loot table?
[latest.log](https://github.com/noobanidus/Lootr/files/10812084/latest.log)

**Comment by noobanidus** [comment]
I should probably clarify further, but there's literally nothing Lootr can do about this. It's an inherent bug in Vanilla.

@Poke097 To clarify, there's nothing I can do to help. I'd suggest seeing if someone is willing to port Async Locator to Fabric.
